custom/icds_reports/utils/aggregation.py

```diff
@@ -1185,14 +1186,25 @@ def aggregation_query(self):
             {calculations}
             FROM "{child_health_case_ucr}" child_health
             LEFT OUTER JOIN "{child_tasks_case_ucr}" child_tasks ON child_health.doc_id = child_tasks.child_health_case_id
+              AND child_health.state_id = child_tasks.state_id
+              AND lower(substring(child_tasks.state_id, '.{{3}}$'::text)) = %(state_id_last_3)s
```

Comment:
These `lower` functions ensure we're using `architect`s check constraints to prune tables. In the future we will probably want to make a change to how it generates these check constraints